### ISSUE TITLE
critic loss

### ISSUE THREAD

**Issue by vince-1998** [EchoRickZz/Pensieve-Pytorch#4]
Hello,I have a question. In your code, in the calculation of critic loss, why do you average 47 video blocks instead of averaging the total critical loss of all agents at the end. Your code is: loss=torch. ean (td_batch. pow (2), axis=0)

**Comment by vince-1998** [comment]
> Hello,I have a question. In your code, in the calculation of critic loss, why do you average 47 video blocks instead of averaging the total critical loss of all agents at the end. Your code is: loss=torch. ean (td_batch. pow (2), axis=0)

Oh, what you do is batch_ Train? I almost misunderstood

**Comment by RuixiaoZhang** [comment]
Hi vince,
Yes, you are right. The critic loss (also actor loss) is optimized through batch training.
